atomicapp/nulecule/container.py

```diff
@@ -31,10 +31,14 @@ def __init__(self, dryrun=False, docker_cli='/usr/bin/docker'):
             except subprocess.CalledProcessError as e:
                 if "client and server don't have same version" in e.output \
                         or "client is newer than server" in e.output:
-                    print("\nThe docker version in this Atomic App differs "
-                          "greatly from the host version.\nPlease use a different "
-                          "Atomic App version for this host.\n")
-                raise e
+                    raise DockerException("\nThe docker version in this Atomic App "
+                                          "differs greatly from the host version.\n"
+                                          "Please use a different Atomic App version "
+                                          "for this host.\n")
+                elif "Is your docker daemon up and running" in e.output or \
+                     "Are you trying to connect to a TLS-enabled daemon without TLS" \
+                     in e.output:
+                    raise DockerException("Could not connect to the docker daemon.")
```

Comment:
So there is a bit of history behind this. See https://github.com/projectatomic/atomicapp/issues/174 and then the PR https://github.com/projectatomic/atomicapp/pull/361. I think the idea was to give the user of atomicapp a message that helped them understand that the version of docker within the atomicapp container differed from the host and what they needed to do to resolve that.

Just spitting out e.output doesn't really help them with that particular problem. If there are problems that could be specific to atomic apps interaction with docker then it would do our users some good to have a more helpful message delivered to them.